clients/socket-client.rst

```diff
@@ -63,4 +63,16 @@ use case would be::
     ];
     $client = new Client($messageFactory, $options);
 
+.. warning::
+
+    This client assumes that the request is compliant with HTTP 2.0, 1.1 or 1.0 standard. So a request without a ``Host`` header, or
+    with a body but without a ``Content-Length`` will certainly fail.
+    To make sure all requests will be sent out correctly, we recommend to use the ``PluginClient`` with the following plugins:
+
+    * ``ContentLengthPlugin`` sets the correct ``Content-Length`` header, or decorate the stream to use chunked encoding
+    * ``DecoderPlugin`` decodes encoding coming from the response (chunked, gzip, deflate and compress)
+    * ``HeaderDefaultsPlugin`` sets default headers values if not present (like for the ``Host`` header)
```

Comment:
erm. HeaderDefaultsHeader just accepts a bunch of default values. and AddHostPlugin is about adding host (and scheme and port) to a path if the request is only for a path.

it sounds like we should come up with a new plugin SetHostHeaderFromUriPlugin or something like that?